vortex-array/src/compute/boolean.rs

```diff
@@ -0,0 +1,39 @@
+use vortex_error::VortexResult;
+
+use crate::{Array, IntoArrayVariant};
+
+pub trait AndFn {
+    fn and(&self, array: &Array) -> VortexResult<Array>;
+}
+
+pub trait OrFn {
+    fn or(&self, array: &Array) -> VortexResult<Array>;
+}
+
+pub fn and(lhs: &Array, rhs: &Array) -> VortexResult<Array> {
+    if let Some(selection) = lhs.with_dyn(|lhs| lhs.and().map(|lhs| lhs.and(rhs))) {
```

Comment:
Maybe leave a couple of comments, these functions are hard to parse at first glance

Reply:
added a couple in #488